### PR TITLE
Remove support for 1.x snapshots

### PR DESCRIPTION
**Pull request by jumaffre**
Addresses 1. in https://github.com/microsoft/CCF/issues/2981

Now that 2.x releases are cut from their own `release/2.x` branch, we no longer need to support 1.x snapshots on `main`. These snapshots didn't embed a receipt because these weren't available at the time snapshots were introduced, but they do now and we do not expect operators to start 3.x services from a 1.x snapshots. 

Note: Also contains changes from #4226

### CODE REVIEW

**Comment by achamayou** [comment]
@jumaffre can we split out the changes from #4226? They make reviewing this PR to check it only changes the snapshot handling a bit difficult.

File: src/node/snapshot_serdes.h [hunk 1]
```diff
@@ -71,62 +50,56 @@ namespace ccf
     auto tx_hdr = serialized::peek<kv::SerialisedEntryHeader>(data, size);
     auto store_snapshot_size = sizeof(kv::SerialisedEntryHeader) + tx_hdr.size;
 
-    // Snapshots without a snapshot evidence seqno specified by the host should
-    // be self-verifiable with embedded receipt
-    if (!evidence_seqno.has_value())
+    auto receipt_data = data + store_snapshot_size;
+    auto receipt_size = size - store_snapshot_size;
+
```
Comment (achamayou) [inline src/node/snapshot_serdes.h:55]: An early check here, with an error message if the receipt_size is empty to point out that this is unsupported snapshot may be good.

**Comment by jumaffre** [comment]
> @jumaffre can we split out the changes from #4226? They make reviewing this PR to check it only changes the snapshot handling a bit difficult.

Yes. The goal was to have #4226 merged first before opening this PR for review. Hopefully #4226 gets a green CI run soon and this PR gets easier to review/merge.

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,10 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 - `set_user` action in sample constitutions correctly handles `user_data` (#4229).
 
+### Removed
+
+- Snapshots generated by 1.x services can no longer be used to join from or recover a new service (#4255).
```
Comment (achamayou) [inline CHANGELOG.md:16]: Let's add a note to make it clear that this means a 1.x ledger must effectively upgrade to 2.x first and produce at least one snapshot, before going to 3.x. Or that it is at the very least strongly recommended, unless the ledger is quite short.

**Review by achamayou — APPROVED** [review]

File: include/ccf/node_startup_state.h [hunk 3]
```diff
@@ -14,8 +14,7 @@ namespace ccf
     partOfPublicNetwork,
     partOfNetwork,
     readingPublicLedger,
-    readingPrivateLedger,
-    verifyingSnapshot
+    readingPrivateLedger
```
Comment (achamayou) [inline include/ccf/node_startup_state.h:17]: That enum is never serialized to the ledger, right?

File: src/kv/kv_types.h [hunk 4]
```diff
@@ -354,7 +354,6 @@ namespace kv
     PASS_BACKUP_SIGNATURE_SEND_ACK = 4,
     PASS_NONCES = 5,
     PASS_NEW_VIEW = 6,
-    PASS_SNAPSHOT_EVIDENCE = 7,
```
Comment (achamayou) [inline src/kv/kv_types.h:357]: Is that gap correctly handled as unknown status? We don't want it falling through accidentally.

File: include/ccf/node_startup_state.h (hunk 3 above)
Comment (jumaffre) [reply include/ccf/node_startup_state.h:17]: It isn't. It's only reported by one of the `/node` operator endpoints.

Comment (achamayou) [reply include/ccf/node_startup_state.h:17]: Great, I thought this was the case, just wanted to confirm.

File: src/kv/kv_types.h (hunk 4 above)
Comment (achamayou) [reply src/kv/kv_types.h:357]: Great, thank you.

**Review by achamayou — APPROVED** [review]

**Pull request merged**